samples/oauth_rails_vethealth/app/models/health_api_response.rb

```diff
@@ -0,0 +1,55 @@
+class HealthApiResponse
+  attr_reader :api_name, :target, :page, :count
+  def initialize(api_name:, id:, access_token:, action: :read, page: 1, count: 10)
+    @api_name, @id, @action = api_name, id, action
+
+    case action
+    when :search
```

Comment:
Why are the actions being managed in the model and not in the controller? You could have the controller handle this logic and then call the appropriate queries on the model

Reply:
I'm not sure what you mean here?  The responses are all used the same, but the target is generated differently based on whether you want to make a "read" request or a "search" request, so I thought that was handled neatly in the initializer.

Reply:
Discussed, concluded that refactoring is too painful here